extensions/sql-migration/src/dashboard/sqlServerDashboard.ts

```diff
@@ -0,0 +1,502 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import { MigrationLocalStorage } from '../models/migrationLocalStorage';
+import * as loc from '../models/strings';
+import { IconPathHelper } from '../constants/iconPathHelper';
+
+interface IActionMetadata {
+	title?: string,
+	description?: string,
+	link?: string,
+	iconPath?: { light: string | vscode.Uri; dark: string | vscode.Uri },
+	command?: string;
+}
+
+const maxWidth = 800;
+
+export class DashboardWidget {
+
+	private _migrationStatusCardsContainer!: azdata.FlexContainer;
+	private _view!: azdata.ModelView;
+
+	constructor() {
+	}
+
+	public register(): Promise<void> {
```

Comment:
From what I can tell nothing is actually waiting on the promise returned so I'd suggest just removing the promise and just have this function call registerModelViewProvider but not wait for it to be fully initialized. Having the stuff wrapped in the promise is just unnecessary currently.